src/aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -146,6 +153,40 @@ def prepare_process(self):
             self.ctx.inputs.parameters['INPUTPH']['recover'] = True
             self.ctx.inputs.parent_folder = self.ctx.restart_calc.outputs.remote_folder
 
+    def create_merged_output(self):
+        """Merge outputs from multiple ``PhCalculation`` runs called by the workchain if necessary."""
+        if self.inputs.only_initialization.value:
+            return
+
+        output_dict = {
+            'output_' + str(index + 1): child.outputs.output_parameters
+            for index, child in enumerate(self.ctx.children)
+        }
+
+        num_qpoints = self.ctx.children[0].outputs.output_parameters['number_of_qpoints']
+        num_qpoints_final = self.ctx.children[-1].outputs.output_parameters['num_q_found']
+
+        if num_qpoints == num_qpoints_final:
+            return
+
+        num_qpoints_found = sum(output['num_q_found'] for output in output_dict.values())
+
+        if num_qpoints_found == num_qpoints:
+            self.report(f'Merging {num_qpoints} q-points data from `PhCalculation`s.')
+            self.ctx.merged_output_parameters = merge_ph_outputs(**output_dict)
+        else:
+            self.report(f'Only {num_qpoints_found} of {num_qpoints} q-points were parsed.')
+            return self.exit_codes.ERROR_MERGING_QPOINTS
```

Comment:
The user could have chosen just to run a few q-points with start_q/last_q and in this case, the number of qpoints parsed from the beginning of the file will be determined by the qmesh and it will appear as if there are q-points missing. Maybe we remove this error, or check if the user specified this input

Reply:
This case is now also considered by checking if either of these inputs have been set and recalculating the `num_qpoints` a few lines above.